freqtrade/optimize/backtesting.py

```diff
@@ -745,15 +747,51 @@ def start(self) -> None:
         self.load_bt_data_detail()
         logger.info("Dataload complete. Calculating indicators")
 
+        run_ids = {
+            strategy.get_strategy_name(): get_strategy_run_id(strategy)
+            for strategy in self.strategylist
+        }
+
+        # Load previous result that will be updated incrementally.
+        if self.timerange.stopts == 0 or datetime.fromtimestamp(
+           self.timerange.stopts, tz=timezone.utc) > datetime.now(tz=timezone.utc):
```

Comment:
I think your own analysis was quite good - this can be worked around eventually (i'm however not sure how to protect from it).

using `timerange` from above as comparison will not protect aganst this:

* backtest with `--timerange 20210101-20220101` (full prior year) with data ending 2021-11-01.
* download data from November until today
* run the same backtest again `--timerange 20210101-20220101`

By comparing the 2 timerange objects (timerange - self.timerange) - you could protect against the first case (not loading cache) - but you'll not know that you had 2 months missing in the "cached" result (nothing in the config or strategy changed - other than the messages at startup).


TBH - i'm not sure we must protect against this ... we can also add a hint in the documentation about this, warning users to use `--no-cache` when the underlying data changed.